### PR TITLE
prometheus-redis-exporter: init at 1.7.0

### PR DESCRIPTION
**Pull request by srhb**
###### Motivation for this change
I needed this for production.

###### Things done
vgo2nix'ed deps pending discussions of https://github.com/NixOS/nixpkgs/issues/84826 to converge on some recommendation, I suppose.

Left out a few of the possible "version" parameters as "unknown" since I feel they're likely to get out of date during updates, and don't provide much additional value (save perhaps for sha1, reviewer might disagree with me there.)

Tested against local and remote redis servers. Stats are served and look correct.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by srhb** [comment]
@eskytthe Added you as a maintainer here as requested. :)

**Review by Ma27 — COMMENTED** [review]
Would you mind adding a module for the prometheus-exporter (and probably a test as well)?

**Comment by srhb** [comment]
Oh, yes, of course. Totally forgot about that. :) On it.

**Comment by srhb** [comment]
Done. The module is very minimal as the exporter is entirely configurable through command line flags . What a nice framework for creating and testing exporters. :)

File: pkgs/servers/monitoring/prometheus/redis-exporter.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  buildFlagsArray = ''
+    -ldflags=
+       -X main.BuildVersion=${version}
+       -X main.BuildCommitSha=unknown
+       -X main.BuildDate=unknown
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Prometheus exporter for Redis metrics";
+    homepage = "https://github.com/prometheus/consul_exporter";
```
Comment (WilliButz) [inline pkgs/servers/monitoring/prometheus/redis-exporter.nix:27]: This looks like a leftover from copy-pasting :)
```suggestion
    inherit (src.meta) homepage;
```

**Review by WilliButz — CHANGES_REQUESTED** [review]
Looks good to me, just one minor thing:

File: pkgs/servers/monitoring/prometheus/redis-exporter.nix (hunk 1 above)
Comment (srhb) [reply pkgs/servers/monitoring/prometheus/redis-exporter.nix]: It was indeed, sorry about that. Fixed. :)

**Review by WilliButz — APPROVED** [review]
:+1:

**Comment by WilliButz** [comment]
@GrahamcOfBorg test prometheus-exporters.redis

**Pull request merged**